### ISSUE TITLE
Collada loading errors with drcsim prerelease from .debs

### ISSUE THREAD

**Issue by osrf-migration** [osrf/drcsim#466]
**Original report ([archived issue](https://osrf-migration.github.io/drcsim-gh-pages/#!/osrf/drcsim/issues/466)) by Stefan Kohlbrecher (Bitbucket: [Stefan_Kohlbrecher](https://bitbucket.org/%7Bdb06b03e-7b31-4483-9018-534df5bd723f%7D/)).**

----------------------------------------
With current drcsim prerelease from .debs installed, the following gets spammed (at least for me) when Gazebo comes up:

```
#!

[Err] [ColladaLoader.cc:1670] Collada file[/opt/ros/indigo/share/drcsim_model_resources/gazebo_models/environments/atlas_description/meshes_unplugged/r_clav.dae] is invalid. Loading what we can...
[Err] [ColladaLoader.cc:1670] Collada file[/opt/ros/indigo/share/drcsim_model_resources/gazebo_models/environments/atlas_description/meshes_unplugged/r_clav.dae] is invalid. Loading what we can...
[Err] [ColladaLoader.cc:1670] Collada file[/opt/ros/indigo/share/drcsim_model_resources/gazebo_models/environments/atlas_description/meshes_unplugged/pelvis.dae] is invalid. Loading what we can...
```

While this wouldn't be expected from above output, the robot model gets spawned just fine in Gazebo. I noticed that the .dae files for most (but not all) Atlas links fail to load on ROS hydro rviz, however. That's another hint that something about them seems to be incompatible with the standard (?).

**Comment by osrf-migration** [comment]
**Original comment by Stefan Kohlbrecher (Bitbucket: [Stefan_Kohlbrecher](https://bitbucket.org/%7Bdb06b03e-7b31-4483-9018-534df5bd723f%7D/)).**

----------------------------------------

* Edited issue description

**Comment by osrf-migration** [comment]
**Original comment by Ian Chen (Bitbucket: [Ian Chen](https://bitbucket.org/%7Beaa6fca5-6deb-43f6-907f-971c144735dd%7D/), GitHub: [iche033](https://github.com/iche033)).**

----------------------------------------
fixed in [pull request #1339](https://bitbucket.org/osrf/gazebo/pull-request/1339/add-checks-for-handling-arrays-with-counts/diff), awaiting a new gazebo 4.x release

**Comment by osrf-migration** [comment]
**Original comment by Steve Peters (Bitbucket: [Steven Peters](https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/), GitHub: [scpeters](https://github.com/scpeters)).**

----------------------------------------
I don't know why rviz would be having trouble, but there is a fix for the warning messages in gazebo, as @iche033 said.

**Comment by osrf-migration** [comment]
**Original comment by Stefan Kohlbrecher (Bitbucket: [Stefan_Kohlbrecher](https://bitbucket.org/%7Bdb06b03e-7b31-4483-9018-534df5bd723f%7D/)).**

----------------------------------------
Ok, sounds good. Not sure about the rviz trouble (it obviously also works in indigo rviz). I think this can be closed then.

For reference in case of future woes with the collada files I´m adding some info about what happens with rviz hydro:

atlas_v4 in rviz hydro:
![atlas_v4_rviz_hydro.png](https://osrf-migration.github.io/drcsim-gh-pages/data/bitbucket.org/repo/bae8oa/images/661200733-atlas_v4_rviz_hydro.png)

atlas_v5 in rviz hydro:
![atlas_v5_rviz_hydro.png](https://osrf-migration.github.io/drcsim-gh-pages/data/bitbucket.org/repo/bae8oa/images/4260216900-atlas_v5_rviz_hydro.png)

As can be seen, for some reason a few collada files get loaded, while others don´t. Interestingly, the same links are affected, even though different models are used. Terminal output:


```
#!

[ERROR] [1420704646.113272418]: Could not load model 'package://atlas_description/meshes_unplugged/utorso.dae' for link 'utorso': OGRE EXCEPTION(6:FileNotFoundException): Cannot locate resource package://atlas_description/meshes_unplugged/utorso.dae in resource group Autodetect or any other group. in ResourceGroupManager::openResource at /build/buildd/ogre-1.7.4/OgreMain/src/OgreResourceGroupManager.cpp (line 753)
[ERROR] [1420704646.113415926]: Could not load model 'package://atlas_description/meshes_unplugged/utorso.dae' for link 'utorso': OGRE EXCEPTION(6:FileNotFoundException): Cannot locate resource package://atlas_description/meshes_unplugged/utorso.dae in resource group Autodetect or any other group. in ResourceGroupManager::openResource at /build/buildd/ogre-1.7.4/OgreMain/src/OgreResourceGroupManager.cpp (line 753)
```

**Comment by osrf-migration** [comment]
**Original comment by Chris Swetenham (Bitbucket: [cswetenham](https://bitbucket.org/%7B481e143e-0abe-4e9d-a094-56d22cdd6626%7D/)).**

----------------------------------------
We are seeing this issue in Hydro; we would much prefer to stay on Hydro until the DRC Finals if possible.

**Comment by osrf-migration** [comment]
**Original comment by Stefan Kohlbrecher (Bitbucket: [Stefan_Kohlbrecher](https://bitbucket.org/%7Bdb06b03e-7b31-4483-9018-534df5bd723f%7D/)).**

----------------------------------------
It might be worth to take a look at the COLLADA files for the l_uleg/r_uleg (i.e. thigh) geometries as the right one curiously shows up, while the left one doesn't do so. Even on Indigo, they're a shaded a little bit differently, suggesting that even there some (likely unintended) difference exists.

**Comment by osrf-migration** [comment]
**Original comment by Carlos Agüero (Bitbucket: [caguero](https://bitbucket.org/%7Bda8a8e89-4bb0-421b-bd0e-dbbed3d4ed6a%7D/), GitHub: [caguero](https://github.com/caguero)).**

----------------------------------------

* changed state from "new" to "resolved"

Merged in fix_dae_rviz ([pull request #524](https://osrf-migration.github.io/drcsim-gh-pages/#!/osrf/drcsim/pull-requests/524))

Fix issue #466

→ \<\<cset 77ff731df55e56b1156cb21414e51f16f6314399\>\>
